ext/spl/spl_fixedarray.stub.php

```diff
@@ -9,6 +9,10 @@ public function __construct(int $size = 0) {}
     /** @tentative-return-type */
     public function __wakeup(): void {}
```

Comment:
`__wakeup` is mostly only called for unserialization, not directly



1. I think this should emit a deprecation notice - once php fully deprecates dynamic properties in php 9.0 there will be even less of a reason to ever call this. Adding an `@deprecated` line to the PHPDoc comment does that, I believe
2. Personally, I think the implementation of `__wakeup` should be changed to do nothing (and emit a deprecation notice), and be removed in the next major version. User code calling `$fixedArray->__wakeup` directly and copying declared properties (not changing them) and moving+removing dynamic properties to the array (but only if the array size is 0) is unintuitive and not something you'd deliberately do. The deprecation notice will tell anything doing that that it will no longer work.
3. The behavior of `get_object_properties{,_for}` has weirdness as a result of what `__wakeup` does right now - thankfully, the impact of some of that will go away with the switch to `__serialize` and the rest is out of scope


```php
php > class MyClass extends SplFixedArray { public function __construct(public readonly int $abc) {} }
php > $x = new MyClass(123);
php > var_export($x);
\MyClass::__set_state(array(
   'abc' => 123,
))
php > echo serialize($x);
O:7:"MyClass":1:{s:3:"abc";i:123;}
php > $x->__wakeup();
php > echo serialize($x);
O:7:"MyClass":1:{i:0;i:123;}  // <-- no longer serializing $x->abc
php > var_export($x->abc);
123
```

Reply:
I also support the deprecation of `__wakeup()`, but it seems a bit late, just before RC1. Of course, it is possible that doing so is still the least worst option. To be honest, I definitely wouldn't remove the whole code of `__wakeup` just yet, deprecating first seems a smoother upgrade path IMO.